### ISSUE TITLE
Arrows / enderpearls crashing clients

### ISSUE THREAD

**Issue by alexmarkey** [MylesIsCool/ViaVersion#248]
### What version of ViaVersion are you using? Type /ver viaversion

0.6.5

### What version of Spigot are you using? Type /ver

This server is running CraftBukkit version git-PaperSpigot-"a86
1cc6" (MC: 1.8.8) (Implementing API version 1.8.8-R0.1-SNAPSHOT)

### What plugins are you using? Type /plugins

Plugins (12): LilyPad-Connect, AutoLapis, Updater, ViaVersion,
HideStream, LilyAlert, WorldReset, NoHitDelay, ProtocolLib, TeamSkywars, NoCheat
Plus, HolographicDisplays

### Are you using any additional software like BungeeCord? If so, what software and version? (Please list your plugins as well)

Using LilyPad, Plugins are LilyPad-Connect

### How does this error happen? login? Using an item? 

Randomly happens to some players when throwing enderpearls or shooting bows throughout our server.

### Is there an error in the console? Use pastebin.com. Is there a kick message?

Kick message - http://prntscr.com/ah1ou6

**Comment by HugoDaBosss** [comment]
So this does not happen all the time? It could be caused by the combination of PaperSpigot and LilyPad, but these are just speculations. Are you changing anything with has to do with shooting projectiles in your TeamSkyWars plugin?

**Comment by alexmarkey** [comment]
Seems to randomly happening,We have explosive arrows in our Skywars / TeamSkywars (Arrow lands creates an explosion), that is the only change we have made to projectiles. 

http://mccentral.org/forum/m/17339729/viewthread/26542873-still-crashing-in-team-skywars-recreated

http://mccentral.org/forum/m/17339729/viewthread/26555425-its-back

**Comment by alexmarkey** [comment]
Issue is not limited to Skywars / Team Skywars, have had reports of it in Survival Games and pretty much everywhere else on our network (Where explosive arrows are not used).

The crashes do not necessarily happen to the player firing the arrow, have had reports where someone pulls a bow and shoots at a player and when they shoot it the player being shot at crashes (Same error)

**Comment by Matsv** [comment]
Do you mean with 'the shot crashes' it doesn't look like the arrow fires?

**Comment by alexmarkey** [comment]
Conversation with one of my players that encountered it - http://pastebin.com/SfW2qcQZ

Have never experienced it myself as I do not play my own server however from what I can gather its when players shoot arrows, any clients within the vicinity of the player shooting the arrow is prone to be crashed.

**Comment by alexmarkey** [comment]
I believe every player that has reported this has been using optifine however optifine is used by a large quantity of Minecrafters so this could just be co-incidence, however as I said, using complete Vanilla Minecraft I have never encountered this issue.

**Comment by Matsv** [comment]
@alexmarkey But optifine is not out yet for 1.9? Is 1.8 crashing?

**Comment by MylesIsCool** [comment]
This will need retesting with latest build from build server to see if the patch fixes it.

**Comment by alexmarkey** [comment]
How long till the official build is released?

**Comment by alexmarkey** [comment]
Going to test the dev build on our minigames tonight, can also add more info that Fishing rods cause this same crash.

**Comment by alexmarkey** [comment]
Alright finally managed to reproduce this:

This is Only reproduceable using LilyPad (I think, When using no lilypad I get no client crashes but I believe this can still be fixed on the ViaVersion side of things)

The arrow issue only happens when the server is FULL of 1.8 clients, if there is 1 player using 1.9 NO clients will crash.

So for example when testing on a server using lilypad I was the only player on the server and I was using 1.8, I crash everytime I shoot an arrow, If I log on an alt account (Using 1.9) then fire an arrow on the 1.8 client it will NOT crash.

So to sum it up, to reproduce the issue, use a lilypad server full of 1.8 clients and fire an arrow and your client will crash.

Because our server has Minigame instances (Skywars arenas with only 6 players in them) this happens to us very regularly.

**Comment by MylesIsCool** [comment]
Closed, related to LilyPad / Map Issue, working with to help find a solution.
